testing/test_collection.py

```diff
@@ -489,7 +489,7 @@ def test_collect_topdir(self, pytester: Pytester) -> None:
         # assert root2 == rcol, rootid
         colitems = rcol.perform_collect([rcol.nodeid], genitems=False)
         assert len(colitems) == 1
-        assert colitems[0].path == p
+        assert colitems[0].path == topdir
```

Comment:
this one seems off - the path there should match the nodeid

Reply:
The old tree was

```
<Session>
  <Module test_collect_topdir.py>
```

Now it's 

```
<Session>
  <Dir test_collect_topdir0>
    <Module test_collect_topdir.py>
```

So now the collection (with `genitems=False`) collects the `Dir` not the `Module`.

I'm not sure what was the original purpose of this test (it goes back to e2683f4538d8faf663be5acc3874051c3486adde) but the change here is intentional. Do you think there is something that relies on this?

Reply:
The purpose of that function is to go from a list of nodeids/paths to the initial collectors

Aka the concrete "top-level" fs collectors that are files

The alternative is to ensure all directory collection is limited to filtering of the initial args